### ISSUE TITLE
Logger cannot merge runs in output data when called out of context

### ISSUE THREAD

**Issue by nojhan** [IOHprofiler/IOHexperimenter#39]
I'm interfacing my solver with IOHexperimenter at the problem level, because I wish to use my own experiment manager. Indeed, I wish to use my own batch system to place (down to) single runs as independent jobs on an arbitrary set of machines. However, this is not feasible with the current architecture.

The core of the problem is that the Logger interface expect to be initialized once, and then to be called on different `target_problem`, for several runs. However, if one wish to handle jobs at the runs level, the Logger will be initialized for each run. In that setting, the Logger avoid adding the results in the previously written data folder, and create an additional folder, leading to two different folders holding two separated runs.

Unfortunately, this data structure cannot be loaded as a single experiment by IOHanalyzer.

To test the problem, merge PR #38 and run `t-bbob-csvlogger` twice.

The solutions I can imagine are to:
- add a state manager to Logger that would allow to load previous data structures from saved files,
- allow IOHanalyzer to parse several folders as a single experiment,
- handle this specific case with an additional data merger.

**Comment by jacobdenobel** [comment]
@nojhan Is this still relevant?

**Comment by nojhan** [comment]
Yes. There is no way to parallelize runs out of a single executable while merging the data of the `logger::Analyzer` in the same folder.

The use case would be to spread the runs on an HPC cluster, while having a single folder with all the data at the end.

Having a script able to merge two folders seems like the easier approach here.

**Comment by jacobdenobel** [comment]
This is available in Python, but not in yet C++.

**Comment by nojhan** [comment]
That's sufficient, given that it's needed out of executables anyway. Is it
documented somewhere?

**Comment by jacobdenobel** [comment]
I think this method does what you need. https://github.com/IOHprofiler/IOHexperimenter/blob/8a49d76d591c52b4ae8ed0991d4b6ea8d5c3adaa/ioh/__init__.py#L378 

If you use the `Experiment` class in Python you can use it. This uses Python multiprocess to parallize, so I think the beviour is similar to what you intend.
